.github/workflows/ci-cd-api-server.yaml

```
@@ -52,26 +53,58 @@ jobs:
+              --desired-count 0
+          
+            # 모든 작업이 종료될 때까지 대기
+            echo "Waiting for tasks to drain..."
+            aws ecs wait services-stable \
+              --cluster ${{ env.ECS_CLUSTER }} \
+              --services ${{ env.ECS_SERVICE }}
+          fi
 
+      # 8. ECS 배포
       - name: Deploy to Amazon ECS
         run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2086:info:11:21: Double quote to prevent globbing and word splitting

(shellcheck)